### DISCUSSION TITLE
[Art competition] Selective widening

### DISCUSSION THREAD

**Discussion by Re-st** [prosyslab-classroom/cs524-program-analysis#281] [Art competition]
![_854c04ae-6fcc-45c3-9d38-94a7de167528](https://github.com/prosyslab-classroom/cs524-program-analysis/assets/42648067/4ddcf7cd-4b22-4d7c-a21d-ef1ab311a338)

## Short Description
For me, [*Selective widening*] means [*detective drawing relations on a chalkboard, realistic style.*]".

## Long Description (up to 3 sentences)
1. Due to limited resources, detectives must distinguish between people to investigate in detail and those to skip.
2. The clock is a metaphor for the scarcity of computing resources, while the graph drawn by the detective represents cycle-detection for the further act of widening for non-suspects. 
3. Focusing on crucial parts (through joining, in selective widening) is for getting a contour of the case in limited time, which is important in both crime investigation and static analysis.

The whole prompt is "Detective drawing relations on a chalkboard, realistic style. Wall-clock is juxtaposed with the calendar, a scratchboard full of evidence post-its, and the chalkboard. Size of each object is realistic. Detective should take at least 20% of the picture. Detective wears typical detective attire, and is fully visible."

**Comment by KihongHeo** [comment]
Nice try!
